src/cysignals/implementation.c

```diff
@@ -107,6 +108,60 @@ static inline void reset_CPU(void)
 }
 
 
+/* Reset all signal handlers and the signal mask to their defaults. */
+static inline void sig_reset_defaults(void) {
```

Comment:
Given that calls to `sig_reset_defaults` and `sigdie_for_sig` always appear together, why make them separate functions? They are logically related because I need to reset the signal handlers for the `kill` in `sigdie` to work.

Reply:
Okay--I was sort of split on that because `sig_reset_defaults` looked like it could be useful on its own as well.  But in practice it only ended being used in this one place.  Alternatively I could keep `sig_reset_defaults` as a separate routine, but place a call to it at the beginning of `sigdie`.